#### Import the essential librairies such numpy, plot, sklearn

In [1]:
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt
from sklearn.datasets import load_iris

##### Load the Iris flowers dataset And prepare the dataFrame

In [2]:
dataset = load_iris()

In [3]:
features, targets = dataset.data, dataset.target
dataset.feature_names.append("category")
names=dataset.feature_names

In [4]:
dataset = pd.concat([pd.DataFrame(features), pd.Series(targets)], axis=1, names=names );dataset.columns = names

In [5]:
dataset.sample(3)

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)  \
87                 6.3               2.3                4.4               1.3   
146                6.3               2.5                5.0               1.9   
41                 4.5               2.3                1.3               0.3   

     category  
87          1  
146         2  
41          0

#### Check for dataset data types and visualize them depending on the target

In [6]:
target, features = dataset["sepal length (cm)"], dataset.drop("sepal length (cm)", axis=1)

In [7]:
features.dtypes, target.dtypes

(sepal width (cm)     float64
 petal length (cm)    float64
 petal width (cm)     float64
 category               int64
 dtype: object,
 dtype('float64'))

In [ ]:
colors = ["teal", "red", "blue", "green" ]
plt.figure(figsize=(24, 18))
for i in range(4):
    plt.subplot(2, 2, i+1)
    plt.scatter(features[features.columns[i]], target, c=colors[i])
    plt.title(f"{features.columns[i]} / {target.name}")
    plt.xlabel(f"{features.columns[i]}")
    plt.ylabel(f"{target.name}")
#     plt.legend()
# plt.show()
    

###### A voir le Plot ci-dessus on peut dire:
- **En ce qui concerne la longueure du sepal;** _Il n'y a pas de tres grande distrubtion entre longueur et la largeur du sepal_: 
    - c'est-a-dire que la distrution est aleatoire et on ne pourait pas etablir une relation entre la longueur et la largeur du sepal.
    - Neamoins, on vois que la taille du sepal augment pour les fleurs dont la longueur du sepal est compris entre __2.5cm et 3.5cm__ . Ce fait pourrait nous interresse pour plutard

- **En ce qui concerne la largeur du petal; on voit une certaine correlation**:
    - En effet **30% a 40%** des fleurs ont une largeur de sepal compris entre **4.0cm et 6.0cm**
    - Le reste des fleur, c'est a dire plus de **55%** correlent avec la largeur de sepal
    
- **En ce qui concerne la longueur du petal:**
     - Ils ont presque la meme corelation que la __largeur du petal__ car pres de **30 a 40%** sont stack a une largeur compris entre **4.5 et 6.0cm**
     - Le reste des fleurs sont elles aussi en correlation avec la largeur du sepal
     - Ce qui constitue une difference entre la largeur et la longueur du sepal, c'est la dispersion des element qui les constitue. On voit au niveau de la longueur que les fleurs sont **"sparse"** contrairement aux autres
     
- **En ce qui concerne les category:**
    - Il n'y a pas de relation. Donc, nous verrons cela comme feature a analyser plutard.

- **Parceque le dataset est de taille minume, nous allons vite passer au modeling voir, si on peut composer quelque chose avant de revenir voir si nous n'avons pas omis d'explorer quelque chose d'autre**